securedrop_client/conversation/transcript/transcript.py

```diff
@@ -0,0 +1,42 @@
+from typing import List, Optional
+
+from securedrop_client import db as database
+
+from .items import Item
+from .items import transcribe as transcribe_item
+
+
+def transcribe(record: database.Base) -> Optional[Item]:
+    return transcribe_item(record)
+
+
+_ENTRY_SEPARATOR = "------\n"
+
+
+class Transcript:
+    def __init__(self, conversation: database.Source) -> None:
+
+        self._items = [transcribe(record) for record in conversation.collection]
+
+    def __str__(self) -> str:
+        if len(self._items) <= 0:
+            return "No messages."
+
+        entries: List[str] = []
+
+        context: Optional[str] = None
+
+        for item in self._items:
+            if item is None:
+                continue
+
+            if context is not None and context == item.context:
+                entry = item.transcript
+            else:
+                entry = f"{item.context}{item.transcript}"
```

Comment:
Because `item.context` and `item.transcript` are translated strings, this is a no-no. (Concatenating pieces of text without locale awareness leads to tears.)

My conclusion of that is that all the formatting and translation must happen in this (`conversation.Transcript`) class. Another way to say the same: the "transcript template" belongs in this class. (We can now think about what formats are the most convenient for such a template.)

Reply:
And as a corollary: the different `Item` methods should only return data that can be treated opaquely by the template.

Reply:
Regarding formats for templates: my mind goes instinctively towards a collection of functions in the form of:

```python
def message_template(opaque_data_strings) -> str:
    return _("Some translated chrome: {data}").format(data=opaque_data_strings)
```